### PR TITLE
Add badgekit api mocking

### PR DESCRIPTION
**Pull request by audy**
I have mocking working with the nock library. I can mock `Client`'s calls to the badgekit API.

Sending PR now. Merging with master breaks it. I will work on fixing the merge and come back.

### CODE REVIEW

**Comment by acabunoc** [comment]
sorry, there was a ton of changes with the last pr merged

**Comment by audy** [comment]
I ran `npm lint` but now everything is indented by 4 spaces. Is that intentional?

**Comment by audy** [comment]
Tests were failing because the way I was initializing config did not work with Habitat.

I've added a basic API test using Nock. I still need test a.k.a "golden" data.

**Comment by audy** [comment]
solved the lint issue.

**Comment by acabunoc** [comment]
@audy resolved the 4 space lint issue?

I'll load some data from @akenall later today -- maybe a few authors from a couple papers. How should I let you know about the data?

**Comment by audy** [comment]
@acabunoc yep. good for review.

**Comment by audy** [comment]
> How should I let you know about the data?

I guess let's make a separate issue for testing all API endpoints.

File: test/api.js [hunk 1]
```diff
@@ -1,17 +1,44 @@
 var request = require('supertest');
 
-// provide a session secret for testing
-var config = {
-  SESSION_SECRET: 'test_secret'
-};
+// test config (need to override .env and .env.dist)
```
Comment (acabunoc) [inline test/api.js]: What badgekit are we planning to test on? Not http://badgekit-api-sciencelab.herokuapp.com/? Ohh, were you asking me to setup a separate badgekit-api with "golden" data? Sorry I think I missed all that.

Can we test against our current badgekit server? We can use the defaults in env.dist (except BADGES_SECRET, I guess)

**Comment by acabunoc** [comment]
Thanks @audy this whole setup works well for me -- just a q on what badgekit server we're testing on :)

**Comment by audy** [comment]
@acabunoc I'm not testing against any server. I'm using the nock library which creates a mock server by intercepting http requests to example.com. You can specify the responses.

If you want to setup a sandbox/testing server, I can use that instead. I just like this approach because it works offline and doesn't depend on someone maintaining the test server.

**Comment by acabunoc** [comment]
Ah, that makes sense. Thanks for the explanation :)

**Comment by acabunoc** [comment]
r+! Nice job

**Pull request merged**